src/test/java/hudson/tasks/junit/pipeline/JUnitResultsStepTest.java

```diff
@@ -327,17 +365,22 @@ public boolean apply(@Nullable FlowNode input) {
     }
 
     public static void assertBranchResults(WorkflowRun run, int suiteCount, int testCount, String branchName, String stageName) {
-        assertBranchResults(run, suiteCount, testCount, -1, branchName, stageName);
+        assertBranchResults(run, suiteCount, testCount, -1, branchName, stageName, null);
     }
 
-    public static void assertBranchResults(WorkflowRun run, int suiteCount, int testCount, int failCount, String branchName, String stageName) {
+    public static void assertBranchResults(WorkflowRun run, int suiteCount, int testCount, int failCount, String branchName, String stageName,
+                                           String innerStageName) {
         FlowExecution execution = run.getExecution();
         DepthFirstScanner scanner = new DepthFirstScanner();
         FlowNode aBranch = scanner.findFirstMatch(execution, branchForName(branchName));
         assertNotNull(aBranch);
         TestResult branchResult = assertBlockResults(run, suiteCount, testCount, failCount, aBranch);
+        String namePrefix = stageName + " / " + branchName;
```

Comment:
_nit_ You could have just made `String branchName, String stageName, String innerStageName` into a `String... names`
and then just `Functions.joinPath(names)`

Reply:
Except the order there would need to be `stageName`, `branchName`, `innerStageName`, and I need `branchName` on its own above too. I'll come through and refactor my test utilities here eventually, though.

Reply:
ok, hence me approving beforehand ;)